testutil/base.go

```diff
@@ -60,3 +61,32 @@ func (s *BaseTest) TearDownTest(c *check.C) {
 func (s *BaseTest) AddCleanup(f func()) {
 	s.cleanupHandlers = append(s.cleanupHandlers, f)
 }
+
+// BackupBeforeMocking backups the specified list of elements before further mocking
+func BackupBeforeMocking(mockablesByPtr ...interface{}) (restore func()) {
+	backup := backupMockables(mockablesByPtr)
+
+	return func() {
+		for idx, ptr := range mockablesByPtr {
```

Comment:
nitpick:  golang naming conventions tend to prefer shorter name for local variables, so "i" would be more typical than "idx"

Reply:
Good to know. Thank you.